File: Romans_Empyre/solver.cpp
#include <bits/stdc++.h>
using namespace std;

string decrypt(string encrypted, string chars, int key) {
    string decrypted;
    for (int i = 0; i < encrypted.size(); i++) {
        int origin_index = (chars.find(encrypted.at(i)) - key + chars.size()) % chars.size();
        decrypted += chars.at(origin_index);
    }
    return decrypted;
}

int main() {
    string encrypted = "TEWGEP6a9rlPkltilGXlukWXxAAxkRGViTXihRuikkos";
    string alphabets = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789{}_-.,/%?$!@#";

    for (int i = 1; i <= alphabets.size(); i++) {
        string results = decrypt(encrypted, alphabets, i);
        cout << "Key " << i << ": " << results << endl;
    }
}

File: Romans_Empyre/solver.py
import string

def decrypt(encrypted, chars, key):
    decrypted = []
    for c in encrypted:
        origin_index = (chars.index(c) - key) % len(chars)
        print(origin_index, end=" ")
        decrypted.append(chars[origin_index])
    return "".join(decrypted)

def main():
    encrypted = "TEWGEP6a9rlPkltilGXlukWXxAAxkRGViTXihRuikkos"
    alphabets = string.ascii_letters + string.digits + "{}_-.,/%?$!@#"
    for i in range(1, len(alphabets)):
        flag = decrypt(encrypted, alphabets, i)
        print(f"Key {i}: {flag}")

if __name__ == '__main__':
    main()
    